components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -218,8 +219,26 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
         // Update the collection date: we don't really care if we have data or not, let's
         // always update the sent date.
         updateSentDate(getISOTimeString(now, truncateTo = TimeUnit.Day))
-        // Reschedule the collection.
-        schedulePingCollection(now, sendTheNextCalendarDay = true)
+
+        // Reschedule the collection if we are in the foreground so that any metrics collected after
+        // this are sent in the next window.  If we are in the background, then we may stay there
+        // until the app is killed so we shouldn't reschedule unless the app is foregrounded again
+        // (see GleanLifecycleObserver).
+        if (isApplicationInForeground()) {
+            schedulePingCollection(now, sendTheNextCalendarDay = true)
+        }
+    }
+
+    /**
+     * Helper function to determine whether the app is running in the background or not.
+     *
+     * @return [Boolean] true if running in background, or false if in foreground.
+     */
+    private fun isApplicationInForeground(): Boolean {
+        val procInfo = ActivityManager.RunningAppProcessInfo()
+        ActivityManager.getMyMemoryState(procInfo)
+
+        return procInfo.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND
```

Comment:
This is an **amazing** API from Android.

Reply:
😄 	I hope you are being sarcastic because just looking at the docs for this made me feel wrong, from weird `out` parameter calling structure to the methods and riddled with deprecated `Int` constants representing the importances (all with arbitrary mysterious values). 

The positive side is that it seems to work quite well.

Reply:
It definitely smells of very early single digit Android API

Reply:
Mh, looks like this usage of their API has been [actively discouraged by Google devs themselves](https://groups.google.com/forum/#!msg/android-developers/zH-2bovZSLg/L2YM8Z1N-HwJ). Looks like the only/proper solution to this is to check for `ON_START`.

Reply:
Also: The importance of the process can change, right? At process start it may be something lower but later become FOREGROUND? Also how does that work with other, specifically higher, levels, e.g. IMPORTANCE_FOREGROUND_SERVICE - the app can have foreground UI and a foreground service in the same process (e.g. that happens when media is playing).

Reply:
In light of your concerns, I've decided to switch to using a flag that is set in the `GleanLifecycleObserver` when it handles the lifecycle events.